node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -66,6 +66,19 @@ const COST_WRONG_PARA: Rep = Rep::Malicious("A collator provided a collation for
 const COST_UNNEEDED_COLLATOR: Rep = Rep::CostMinor("An unneeded collator connected");
 const BENEFIT_NOTIFY_GOOD: Rep = Rep::BenefitMinor("A collator was noted good by another subsystem");
 
+/// Time after starting a collation download from a collator we will start another one from the
+/// next collator.
+/// even if the upthe next validator,load was not finished yet.
+///
+/// This is to protect from a single slow collator preventing collations from happening.
+///
+/// With a collation size of 5Meg and bandwidth of 500Mbit/s (requirement for Kusama validators),
+/// the transfer should be possible within 0.1 seconds. 300 milliseconds should therefore be
```

Comment:
We already knew that these calculations are not correct. Otherwise we would not have required to implement this feature of only uploading one collation at a time.

Reply:
Yeah it is still guess work. That's why I added debug logs when the timeouts hit, so we can see how it goes in practice. Previously the calculations were wrong, because they did not take into account concurrent fetches and multiple heads at all. This time I made it 3 times what it should take for one head, so even with 3 heads it should finish before the next download starts. Either way, made it more conservative now and picked 400ms ... in the end we have to test in practice. Just had to justify the numbers somehow.